Sources/Core/AST/AST.swift

```diff
@@ -20,12 +20,12 @@ public struct AST {
     /// The module containing Hylo's core library, if any.
     public var coreLibrary: ModuleDecl.ID?
 
-    /// Information about the compiler processing `self`.
-    public let compiler: CompilerConfiguration
+    /// Conditional compilation factors for processing `self`.
+    public let factors: ConditionalCompilationFactors
 
-    /// Creates an empty AST for given compiler.
-    public init(for compiler: CompilerConfiguration) {
-      self.compiler = compiler
+    /// Creates an empty AST for given conditional compilation factors.
```

Comment:
```suggestion
    /// Creates an empty instance for given conditional compilation factors.
```

The storage is not the AST.